lib/pbench/server/api/resources/server_configuration.py

```diff
@@ -0,0 +1,152 @@
+            ApiSchema(
+                API_METHOD.GET,
+                API_OPERATION.READ,
+                uri_schema=Schema(
+                    Parameter("key", ParamType.KEYWORD, keywords=ServerConfig.KEYS)
+                ),
+                authorization=API_AUTHORIZATION.NONE,
+            ),
+            always_enabled=True,
+        )
+
+    def _get(self, params: ApiParams, request: Request) -> Response:
+        """
+        Get the values of server configuration parameters.
+
+        Args:
+            params: API parameters
+            request: The original Request object containing query parameters
+
+        GET /api/v1/server/configuration?key=config1&key=config2
+        """
+
+        key = params.uri.get("key")
+        try:
+            if key:
+                s = ServerConfig.get(key)
+                return jsonify({key: s.value if s else None})
+            else:
+                return jsonify(ServerConfig.get_all())
+        except ServerConfigError as e:
+            raise APIAbort(HTTPStatus.INTERNAL_SERVER_ERROR, str(e))
+
+    def _put(self, params: ApiParams, _) -> Response:
+        """
+        Set or modify the values of server configuration keys.
+
+        PUT /api/v1/server/configuration
+        {
+            "config": {
+                "dataset-lifetime": 10,
+                "server-state": "running"
+            }
+        }
+
+        PUT /api/v1/server/configuration/dataset-lifetime?value=10
+
+        PUT /api/v1/server/configuration/dataset-lifetime
+        {
+            "value": "10"
+        }
+        """
+
+        try:
+            key = params.uri["key"]
+            try:
+                value = params.query["value"]
+            except KeyError:
+                try:
+                    value = params.body["value"]
+                except KeyError:
+                    raise APIAbort(
+                        HTTPStatus.BAD_REQUEST,
+                        "Incorrect value specification for key {key!r}",
+                    )
+            try:
+                ServerConfig.set(key=key, value=value)
+                return jsonify({key: value})
+            except ServerConfigBadValue as e:
+                raise APIAbort(HTTPStatus.BAD_REQUEST, str(e))
+            except ServerConfigError as e:
+                raise APIAbort(HTTPStatus.INTERNAL_SERVER_ERROR, str(e))
+        except KeyError:
+            pass
+
+        try:
+            config = params.body["config"]
+        except KeyError:
+            raise MissingParameters(["config"])
+
+        failures = []
+        for k, v in config.items():
+            try:
+                ServerConfig.set(key=k, value=v)
+            except ServerConfigError as e:
+                self.logger.warning("Unable to update key {} = {!r}: {}", k, v, str(e))
+                failures.append(k)
+        if failures:
+            raise APIAbort(HTTPStatus.INTERNAL_SERVER_ERROR)
+        return jsonify(ServerConfig.get_all())
```

Comment:
Hmm; returning everything seemed more convenient since I already had the `get_all`; but you're right that it would be more consistent to collect the actual key/value responses from `ServerConfig.set`... and that would avoid the extra call.